core/src/nats.rs

```diff
@@ -421,11 +449,22 @@ impl TypedNats {
     where
         T: TypedMessage,
     {
-        let result = self
+        let bytes = Bytes::from(serde_json::to_vec(value)?);
+        let result1 = self
             .nc
-            .request(value.subject(), Bytes::from(serde_json::to_vec(value)?))
-            .await
-            .to_anyhow()?;
+            .request(value.subject(), bytes.clone());
+        
+        let result = if let Some(tmp_alt_subject) = value.tmp_alt_subject() {
+            let result2 = self.nc
+                .request(tmp_alt_subject, bytes);
+
+            select! {
+                result = result1 => result.to_anyhow()?,
+                result = result2 => result.to_anyhow()?,
+            }
```

Comment:
Not sure I'm following exactly how this `select!` and the `result = ... => ...` parts are working

Reply:
The syntax is specific to the `select!` macro but looks like a `match` statement. Whichever of `result1` or `result2` resolves first becomes the branch that gets executed. In both branches, the result gets bound to `result` (because we have `result =` in both branches) and returned (after calling `.to_anyhow()?` in both cases.